cellmlmanip/_singularity_fixes.py

```diff
@@ -0,0 +1,234 @@
+"""
+`remove_fixable_singularities` specifies a method to remove fixable singularities from a model
+This represents a flattened CellML model and metadata about its variables.
+"""
+from math import isclose
+
+import networkx as nx
+from sympy import (
+    Abs,
+    Add,
+    Eq,
+    Float,
+    I,
+    Mul,
+    Piecewise,
+    Pow,
+    Wild,
+    exp,
+    log,
+    solveset,
+)
+
+import cellmlmanip
+from cellmlmanip.model import FLOAT_PRECISION
+
+
+ONE = cellmlmanip.Quantity(1.0, 'dimensionless')
+
+
+def _generate_piecewise(vs, ve, sp, ex, V):
```

Comment:
This order is quite weird: 3 values of V, then ex, then V again. Maybe ex, V, sp, vs, ve ?
And could they have nicer names, e.g. Vmin or Vlo and Vhi or something?
Also not sure why we need `sp` here (see below)